include/libdnf/common/weak_ptr.hpp

```diff
@@ -33,36 +34,38 @@ struct WeakPtr;
 /// WeakPtrGuard is a resource guard. WeakPtr instances register itself to the resource guard.
 /// And the resource guard invalidates the registered WeakPtrs when the resource is unusable
 /// (eg. its dependecny was released).
+/// Note on thread safety:
+/// Destroying the WeakPtrGuard while simultaneously using its WeakPtrs in another thread is not safe
+/// and can still trigger a race condition.
+// TODO(jrohel): We want thread-safety. Locking isn't for free. Idea is locking in more upper level. Future.
```

Comment:
Why keep this TODO? It seems outdated now, if you want to keep it, how about being more specific about what still needs to be done for thread safety here?

Reply:
I removed TODO.